include/alpaka/warp/WarpUniformCudaHipBuiltIn.hpp

```diff
@@ -0,0 +1,162 @@
+
+#include <alpaka/core/BoostPredef.hpp>
+
+#if defined(ALPAKA_ACC_GPU_CUDA_ENABLED) && !BOOST_LANG_CUDA
+    #error If ALPAKA_ACC_GPU_CUDA_ENABLED is set, the compiler has to support CUDA!
+#endif
+
+#if defined(ALPAKA_ACC_GPU_HIP_ENABLED) && !BOOST_LANG_HIP
+    #error If ALPAKA_ACC_GPU_HIP_ENABLED is set, the compiler has to support HIP!
+#endif
+
+#include <alpaka/core/Unused.hpp>
+#include <alpaka/warp/Traits.hpp>
+
+#include <cstdint>
+
+namespace alpaka
+{
+    namespace warp
+    {
+        //#############################################################################
+        //! The GPU CUDA/HIP warp.
+        class WarpUniformCudaHipBuiltIn : public concepts::Implements<ConceptWarp, WarpUniformCudaHipBuiltIn>
+        {
+        public:
+            //-----------------------------------------------------------------------------
+            WarpUniformCudaHipBuiltIn() = default;
+            //-----------------------------------------------------------------------------
+            __device__ WarpUniformCudaHipBuiltIn(WarpUniformCudaHipBuiltIn const &) = delete;
+            //-----------------------------------------------------------------------------
+            __device__ WarpUniformCudaHipBuiltIn(WarpUniformCudaHipBuiltIn &&) = delete;
+            //-----------------------------------------------------------------------------
+            __device__ auto operator=(WarpUniformCudaHipBuiltIn const &) -> WarpUniformCudaHipBuiltIn & = delete;
+            //-----------------------------------------------------------------------------
+            __device__ auto operator=(WarpUniformCudaHipBuiltIn &&) -> WarpUniformCudaHipBuiltIn & = delete;
+            //-----------------------------------------------------------------------------
+            ~WarpUniformCudaHipBuiltIn() = default;
+        };
+
+        namespace traits
+        {
+            //#############################################################################
+            template<>
+            struct GetSize<
+                WarpUniformCudaHipBuiltIn>
+            {
+                //-----------------------------------------------------------------------------
+                __device__ static auto getSize(
+                    warp::WarpUniformCudaHipBuiltIn const & /*warp*/)
+                {
+                    return warpSize;
+                }
+            };
+
+            //#############################################################################
+            template<>
+            struct All<
+                WarpUniformCudaHipBuiltIn>
+            {
+                //-----------------------------------------------------------------------------
+                __device__ static auto all(
+                    warp::WarpUniformCudaHipBuiltIn const & warp,
+                    std::int32_t predicate)
+                {
+#if defined(ALPAKA_ACC_GPU_CUDA_ENABLED)
+                    auto const mask = static_cast<std::uint32_t>(activemask(warp));
```

Comment:
BUG: do NOT cast the active mask to 32Bit, this will fail on AMD hardware where the active mask is 64bit.

Reply:
This branch is for CUDA only tho?

Reply:
ohh I missed it, you are right!